### PR TITLE
Reduce unnecessary work

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: src/Analyser/ExpressionTypeHolder.php [hunk 1]
```diff
@@ -35,15 +35,15 @@ public function equals(self $other): bool
 
 	public function and(self $other): self
```
Comment (staabm) [inline src/Analyser/ExpressionTypeHolder.php:36]: this method is on the hot path, prevent unncessary function/method call overhead (we do already similar stuff in other methods of this class)

File: src/Analyser/MutatingScope.php [hunk 2]
```diff
@@ -3143,10 +3143,12 @@ public function enterCatchType(Type $catchType, ?string $variableName): self
 			return $this;
 		}
 
+		$intersectedCatchType = TypeCombinator::intersect($catchType, new ObjectType(Throwable::class));
```
Comment (staabm) [inline src/Analyser/MutatingScope.php:3146]: prevent building the same type twice

**Comment by ondrejmirtes** [comment]
Please share Blackfire profile comparison.

**Comment by staabm** [comment]
unable to find a measurable substantial repro

**Pull request closed without merging**